File: pset6/credit/credit.py
from cs50 import get_string
from math import floor

s=get_string("Number: ")
oddsum=0
evensum=0
length=len(s)

print("odds")

for i in range(-2, -(length+1), -2):
	tensdigit=floor((2*int(s[i]))/10)
	onesdigit=(2*int(s[i]))%10
	oddsum+=tensdigit+onesdigit
	print("index: "+str(i)+" is "+ s[i], "which multiplied by two is "+str(tensdigit)+ " and "+ str(onesdigit))

print("evens")
for j in range(-1, -(length+1), -2):
	evensum+=int(s[j])
	print("index: "+str(j)+" is "+ s[j])
print (f"oddsum {oddsum}")
print (f"evensum {evensum}")

if((evensum+oddsum)%10!=0):
	print("INVALID")
else:
	if length==13 and s[0]=="4":
		print("VISA")
	
	elif length==15:
		if s[0]=="3":
			if s[1]=="4" or s[1]=="7":
				print("AMEX")
	
	elif length==16:
		if s[0]=="4":
			print("VISA")
		elif int(s[0])>50 and int(s[0])<56:
			print("MASTERCARD")
	else:
		print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main (void){
    long cardnum = get_long("Number: ");
    int cardlegnth = 1;
    long legnthfinder = cardnum;

    while (legnthfinder>10){
        legnthfinder = legnthfinder/10;
        cardlegnth++;
    }
    
    int doubled = 0;
    int singled = 0;
    
    for (int i = 1; i<cardlegnth; i=i+2)
    {
        long temp = 1;
        
        for(int j = 0; j<i; j++){
            temp = 10*temp;
        }
        if(2*((cardnum/temp)%10)>9){
         doubled = doubled+((2*((cardnum/temp)%10))/10);
         doubled = doubled+(2*((cardnum/temp)%10)%10); 
        }
        else{
        doubled = doubled+2*((cardnum/temp)%10);
        }
    }
   
    for (int i = 0; i<cardlegnth; i=i+2)
    {
        long temp = 1;
        
        for(int j = 0; j<i; j++){
            temp = 10*temp;
        }
        singled = singled+((cardnum/temp)%10);
    }
    long divd = 1;
    for(int m = 0; m<13;m++){
        divd = divd*10;
    }
        if((singled+doubled)%10==0){
            if(cardlegnth==15){
                if(cardnum/divd==34||cardnum/divd==37){
                    printf("AMEX\n");
                }
                else{
                    printf("INVALID\n");
                }
            }
            else if(cardlegnth==16){
                if(cardnum/(divd*10)>50&&cardnum/(divd*10)<56){
                    printf("MASTERCARD\n");
                }
                else if(cardnum/(divd*100)==4){
                    printf("VISA\n");
                }
                else{
                    printf("INVALID\n");
                }
            }
            else if(cardlegnth==14){
                if(cardnum/(divd)==4){
                    printf("VISA\n");
                }
                else{
                    printf("INVALID\n");
                }
            }
            else{
                printf("INVALID\n");
            }
        }
        else{
            printf("INVALID\n");
        }

}